File: app/Harmonize/Harmonize.Client/Model/Transfer/TransferProgress.cs
using System;
using System.Text.Json.Serialization;
namespace Harmonize.Client.Model.Transfer;


public enum TransferDestination
{
    MediaSystem = 0
}

public class TransferProgress
{
    public required Guid MediaEntryId { get; set; }
    public required string Name { get; set; } = string.Empty;
    public required TransferDestination Destination { get; set; }
    public required float Progress { get; set; }
    public string ProgressFmt => $"{Progress}%";
    public required DateTime StartTime { get; set; }
}

File: server/harmonize/harmonize/defs/transferprogress.py
import datetime
import uuid
from enum import Enum

from harmonize.defs.response import BaseSchema


class TransferDestination(Enum):
    MEDIA_SYSTEM = 0


class TransferProgress(BaseSchema):
    media_entry_id: uuid.UUID
    name: str
    destination: TransferDestination
    progress: float
    start_time: datetime.datetime
